File: medplum-typescript/packages/sdk/src/model/fhir/DetectedIssue.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { DetectedissueSeverity } from "./DetectedissueSeverity";
import { DetectedIssue_Evidence } from "./DetectedIssue_Evidence";
import { DetectedIssue_Mitigation } from "./DetectedIssue_Mitigation";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface DetectedIssue {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    status: code | null | undefined;
    code: CodeableConcept | null | undefined;
    severity: DetectedissueSeverity | null | undefined;
    patient: Reference | null | undefined;
    identifiedDateTime: string | null | undefined;
    identifiedPeriod: Period | null | undefined;
    author: Reference | null | undefined;
    implicated: Reference[] | null | undefined;
    evidence: DetectedIssue_Evidence[] | null | undefined;
    detail: string | null | undefined;
    reference: uri | null | undefined;
    mitigation: DetectedIssue_Mitigation[] | null | undefined;
}

File: medplum-java/medplum-java-model/src/main/java/com/fhir/DetectedIssue.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableDetectedIssue.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface DetectedIssue {
  Optional<uri> implicitRules();

  Optional<List<Extension>> modifierExtension();

  Optional<List<ResourceList>> contained();

  Optional<Period> identifiedPeriod();

  Optional<uri> reference();

  Optional<Meta> meta();

  Optional<CodeableConcept> code();

  Optional<Reference> patient();

  Optional<List<DetectedIssue_Mitigation>> mitigation();

  Optional<code> status();

  Optional<String> identifiedDateTime();

  Optional<List<Extension>> extension();

  Optional<List<Reference>> implicated();

  Optional<String> detail();

  Optional<List<Identifier>> identifier();

  Optional<Narrative> text();

  Optional<id> id();

  Optional<code> language();

  Optional<List<DetectedIssue_Evidence>> evidence();

  String resourceType();

  Optional<DetectedissueSeverity> severity();

  Optional<Reference> author();

  static ImmutableDetectedIssue.ResourceTypeBuildStage builder() {
    return ImmutableDetectedIssue.builder();
  }
}
